docs/cloud-workload-security/agent_expressions.md

```diff
@@ -85,6 +85,4023 @@ SECL expressions support several platforms. You can use the documentation below
 * [Linux][2]
 * [Windows][3]
 
+<<<<<<< HEAD
 [1]: /security/threats/workload_security_rules/custom_rules
 [2]: /security/threats/linux_expressions
-[3]: /security/threats/windows_expressions
\ No newline at end of file
+[3]: /security/threats/windows_expressions
+=======
+{{< code-block lang="javascript" >}}
+open.file.path == "/proc/${process.pid}/maps"
+
+{{< /code-block >}}
+
+List of the available variables:
+
+| SECL Variable         |  Definition                           | Agent Version |
+|-----------------------|---------------------------------------|---------------|
+| `process.pid`         | Process PID                           | 7.33          |
+
+## CIDR and IP range
+CIDR and IP matching is possible in SECL. One can use operators such as `in`, `not in`, or `allin` combined with CIDR or IP notations.
+
+Such rules can be written as follows:
+
+
+{{< code-block lang="javascript" >}}
+dns.question.name == "example.com" && network.destination.ip in [192.168.1.25, 10.0.0.0/24]
+
+{{< /code-block >}}
+
+## Helpers
+Helpers exist in SECL that enable users to write advanced rules without needing to rely on generic techniques such as regex.
+
+### Command line arguments
+The *args_flags* and *args_options* are helpers to ease the writing of CSM Threats rules based on command line arguments.
+
+*args_flags* is used to catch arguments that start with either one or two hyphen characters but do not accept any associated value.
+
+Examples:
+* `version` is part of *args_flags* for the command `cat --version`
+* `l` and `n` both are in *args_flags* for the command `netstat -ln`
+
+
+*args_options* is used to catch arguments that start with either one or two hyphen characters and accepts a value either specified as the same argument but separated by the ‘=’ character or specified as the next argument.
+
+Examples:
+* `T=8` and `width=8` both are in *args_options* for the command `ls -T 8 --width=8`
+* `exec.args_options in [ r"s=.*\\" ]` can be used to detect `sudoedit` was launched with `-s` argument and a command that ends with a `\`
+
+### File rights
+
+The *file.rights* attribute can now be used in addition to *file.mode*. *file.mode* can hold values set by the kernel, while the *file.rights* only holds the values set by the user. These rights may be more familiar because they are in the `chmod` commands.
```

Comment:
```suggestion
You can use the *file.rights* attribute in addition to *file.mode*. *file.mode* can hold values set by the kernel, while *file.rights* only holds the values set by the user. These rights may be more familiar because they are in the `chmod` commands.
```